components/modal/aria-helper.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import { forEach } from 'lodash';
+
+let hiddenElements = [],
+	isHidden = false;
+
+/**
+ * Hides all elements in the body element from screen-readers except
+ * the provided element, script elements and elements that already have
+ * an `aria-hidden="true"` attribute.
+ *
+ * The reason we do this is because `aria-modal="true"` currently is bugged
+ * in Safari, and support is spotty in other browsers overall. In the future
+ * we should consider removing these helper functions in favor of
+ * `aria-modal="true"`.
+ *
+ * @param {Element} unhiddenElement The element that should not be hidden.
+ */
+export function hideApp( unhiddenElement ) {
+	if ( isHidden ) {
+		return;
+	}
+	const elements = document.body.children;
+	forEach( elements, ( element ) => {
+		if (
+			element === unhiddenElement ||
+			element.tagName === 'SCRIPT' ||
```

Comment:
On what basis did you decide script (and not any other node types) should be exempt from `aria-hidden` ?

Reply:
I did a simple DOM inspection after running this function without any additional checks, and having script tags with `aria-hidden="true"` seemed unnecessary, so to limit the amount of DOM operations I decided script tags could be excluded from this operation.

Reply:
@afercia Commented: Well `<script>` elements are not content, they don’t need to be hidden. Also, many of them contain templates for the media UI so better not touch them?